docs/site/migration/models/mixins.md

```diff
@@ -6,8 +6,805 @@ sidebar: lb4_sidebar
+  bottom
+
+{% include code-caption.html content="src/models/note.model.ts" %}
+
+```ts
+import {Entity, model, property} from '@loopback/repository';
+import {addCategoryPropertyMixin} from '../mixins/categoryPropertyMixin';
+
+class TempNote extends Entity {
+  @property({
+    type: 'number',
+    id: true,
+    generated: true,
+  })
+  id?: number;
+
+  @property({
+    type: 'string',
+    required: true,
+  })
+  title: string;
+
+  @property({
+    type: 'string',
+  })
+  content?: string;
+
+  constructor(data?: Partial<TempNote>) {
+    super(data);
+  }
+}
+
+@model()
+export class Note extends addCategoryPropertyMixin(TempNote) {
+  constructor(data?: Partial<Note>) {
+    super(data);
+  }
+}
+
+export interface NoteRelations {
+  // describe navigational properties here
+}
+
+export type NoteWithRelations = Note & NoteRelations;
+```
+
+The required property `category` has now been added to the `Note` model via a
+mixin class factory function.
+
+## LoopBack 3 Approach To Creating A Custom Model Method And Remote Model Method Mixin
+
+The
+[Add a New Model Method And a New Endpoint](./methods.md#add-a-new-model-method-and-a-new-endpoint)
+section of the [Migrating custom model methods](./methods.md) document explains
+how a LoopBack 3 developer can define a custom model method named `findByTitle`
+on the `Note` model, and define a remote method to make it available as a new
+endpoint.
+
+In this section, we will show how a LoopBack 3 developer can define a mixin to
+accomplish this.
+
+In LoopBack 3, a developer is able to create a custom model method/remote method
+mixin by:
+
+- placing the mixin logic in a file in a mixins directory
+- updating the server/model-config.json file with the mixin directory location
+- updating the model's json file to include the mixin's name (and options object
+  or boolean)
+
+### Defining The Model Property Mixin findByTitle.js
+
+The developer defines a custom model method/remote method mixin in
+**common/mixins/findByTitle.js** which adds a custom method `findByTitle` to any
+model, and adds a corresponding remote method definition with path
+`/findByTitle` as well. An options property `returnArgumentName` makes it
+possible to customize the name of the return argument. If it is not specified,
+the return argument of 'items' is used as a default.
+
+{% include code-caption.html content="common/mixins/findByTitle.js" %}
+
+```js
+module.exports = function(Model, options) {
+  const returnArgumentName = options.returnArgumentName
+    ? options.returnArgumentName
+    : 'items';
+
+  Model.remoteMethod('findByTitle', {
+    http: {
+      path: '/findByTitle',
+      verb: 'get',
+    },
+    accepts: {arg: 'title', type: 'string'},
+    returns: {arg: returnArgumentName, type: [Model], root: true},
+  });
+
+  Model.findByTitle = function(title, cb) {
+    var titleFilter = {
+      where: {
+        title: title,
+      },
+    };
+    Model.find(titleFilter, function(err, foundItems) {
```

Comment:
Since your callback is just forwarding the results to `cb`, there is no need to write it and you can pass `cb` directly to `find`.

```suggestion
    Model.find(titleFilter, cb);
```